### PR TITLE
Handles shell history path, with or without spaces.

### PR DESCRIPTION
**Pull request by dimitarvp**
_I use Windows 10 x64, Elixir 1.4.5 and Erlang 20 (not the RC, the official version that got out several days ago)._

Directories with spaces are handled via a workaround, namely the path is converted to the DOS name (8.3 format). When `iex` is started,  the following commands are issued:

```elixir
dir = List.to_string(Application.get_env(:kernel, :shell_history_path))
File.exists?(dir)
```
- For `C:\Users\dimi\.elixir-history`, result is:
  * `"C:\\Users\\dimi\\.elixir-history"`
  * `true` (I previously created the directory)
- For `C:\Users\dimi\_elixir history`, result is:
  * `"C:\\Users\\dimi\\_ELIXI~1"`
  * `true` (I previously created the directory)

The 8.3 filenames are supported quite back to the past so it's IMO a safe workaround. I've been using tricks like that one ever since XP first hit major adoption (circa Q1 2002). Please raise concerns if you have more info than me though, I am not all-knowing. :smile: 

NOTES:

- Don't put double quotes around the directory where `ELIXIR_HISTORY_DIR` is first set! This will result in unusable path being passed to Erlang/Elixir.
- The backslash plus double quotes (where `ERL` is set) was proposed by @josevalim and it's working fine _only_ for paths without spaces, but I wasn't able to find a better way, hence I am using the 8.3 name format. I tried many other options (double double quotes around the path, no quotes at all, backslash and double double quotes, and several others) but Erlang kept complaining. I am no Erlang expert so I cut the fight at the 8.3 name format workaround.
- When `ERL` is used (when starting `elixir.bat`) I purposefully didn't use double quotes there; note they are already put when `ERL` is set. Double double quotes break the startup process.

In short, this looks to be working perfectly right now but is very sensitive to even the smallest changes, hence the long and boring explanations.

**QUESTION**: shall we use an environment variable for the history directory path, and if the variable isn't set, only then to use a hardcoded value?

### CODE REVIEW

**Comment by dimitarvp** [comment]
I went with `USERPROFILE` since the common wisdom in the Windows circles is that it should be used for configurations, while `HOMEDRIVE` + `HOMEPATH` should be a home to user's own media and work documents.

(The fact that 99% of the time both paths are the same seems to be irrelevant to that common wisdom so I went with `USERPROFILE`.)

**Comment by dimitarvp** [comment]
All of the above said, no content appears inside the elixir history directory on my Windows 10 machine.

File: bin/iex.bat [hunk 1]
```diff
@@ -40,7 +40,13 @@ echo ** Options can be passed to the Erlang VM using ELIXIR_ERL_OPTIONS or --erl
 goto end
 
 :run
+@set ELIXIR_HISTORY_DIR=%USERPROFILE%\.elixir-history
```
Comment (josevalim) [inline bin/iex.bat]: Just to double check. Does it work with spaces in the directory name?

**Comment by dimitarvp** [comment]
Yes. You can change the name to a directory with spaces and run the script
on Windows, and it will work.

**Comment by OvermindDL1** [comment]
For note, 8.3 filenames on Windows are *NOT* supported depending on the options that NTFS was set with when the filesystem was selected.  The 8.3 layer can be excluded to allow for greater file indexing capabilities, so you cannot trust that the 8.3 interface will always exist.

**Comment by josevalim** [comment]
@dimitarvp Thanks! I have just noticed after our conversation that the Unix one is also broken. I spent the last hour trying to fix it for Unix and I could not. So I will step back from this feature for now and maybe contribute a shell_history_basename to erlang, that would make all of this much easier. I will close this and my PR and add the docs change directly to master. Thank you for all the help!

**Comment by dimitarvp** [comment]
Perfect. If our exchange gives birth to a better solution down the line,
I'm all for it!

Happy to help. 🙏
